rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -191,6 +191,54 @@ rclcpp::init(int argc, char * argv[])
 #endif
 }
 
+std::vector<std::string>
+rclcpp::remove_ros_arguments(int argc, char const * const argv[])
+{
+  rcl_allocator_t alloc = rcl_get_default_allocator();
+  rcl_arguments_t parsed_args;
+
+  if (rcl_parse_arguments(argc, argv, alloc, &parsed_args) != RCL_RET_OK) {
+    std::string msg = "failed to parse arguments: ";
+    msg += rcl_get_error_string_safe();
+    rcl_reset_error();
```

Comment:
There is a function for this: https://github.com/ros2/rclcpp/blob/947e3f7e677df87db4e4b76bc13c0b8d217fbeb1/rclcpp/include/rclcpp/exceptions.hpp#L100-L119

Reply:
There are other opportunities to use this below too.